catroid/src/test/java/org/catrobat/catroid/test/content/actions/IfOnEdgeBounceActionTest.kt

```diff
@@ -0,0 +1,127 @@
+) {
+
+    private lateinit var sprite: Sprite
+    private lateinit var project: Project
+
+    companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0}")
+        fun parameters() = listOf(
+            arrayOf("NO_BOUNCE", 0f, 0f, 0f, 0f, mapOf(90f to 90f)),
+            arrayOf("TOP_BOUNCE", 0f, TOP_BORDER_POSITION, 0f, BOUNCE_TOP_POSITION,
+                    getExpectedDirections(-90 until 0, 0 until 90, 180)),
+            arrayOf("BOTTOM_BOUNCE", 0f, BOTTOM_BORDER_POSITION, 0f, BOUNCE_BOTTOM_POSITION,
+                    getExpectedDirections(-180 until -90, 90..180, 180)),
+            arrayOf("LEFT_BOUNCE", LEFT_BORDER_POSITION, 0f, BOUNCE_LEFT_POSITION, 0f,
+                    getExpectedDirections(-180 until -90, -90 until 0, 0)),
+            arrayOf("RIGHT_BOUNCE", RIGHT_BORDER_POSITION, 0f, BOUNCE_RIGHT_POSITION, 0f,
+                    getExpectedDirections(0 until 180, 90 until 180, 0)),
+            arrayOf("LEFT_TOP_BOUNCE", LEFT_BORDER_POSITION, TOP_BORDER_POSITION,
+                    BOUNCE_LEFT_POSITION, BOUNCE_TOP_POSITION, mapOf(135f to 135f, -45f to 135f)),
+            arrayOf("RIGHT_TOP_BOUNCE", RIGHT_BORDER_POSITION, TOP_BORDER_POSITION,
+                    BOUNCE_RIGHT_POSITION, BOUNCE_TOP_POSITION, mapOf(-135f to -135f, 45f to -135f)),
+            arrayOf("LEFT_BOTTOM_BOUNCE", LEFT_BORDER_POSITION, BOTTOM_BORDER_POSITION,
+                    BOUNCE_LEFT_POSITION, BOUNCE_BOTTOM_POSITION, mapOf(-135f to 45f, 45f to 45f)),
+            arrayOf("RIGHT_BOTTOM_BOUNCE", RIGHT_BORDER_POSITION, BOTTOM_BORDER_POSITION,
+                    BOUNCE_RIGHT_POSITION, BOUNCE_BOTTOM_POSITION, mapOf(-45f to -45f, 135f to -45f))
+        )
+
+        private const val WIDTH = 100f
+        private const val HEIGHT = 100f
+        private const val SCREEN_WIDTH = 480
+        private const val SCREEN_HEIGHT = 800
+        private const val TOP_BORDER_POSITION = SCREEN_HEIGHT / 2f
+        private const val BOTTOM_BORDER_POSITION = -TOP_BORDER_POSITION
+        private const val RIGHT_BORDER_POSITION = SCREEN_WIDTH / 2f
+        private const val LEFT_BORDER_POSITION = -RIGHT_BORDER_POSITION
+        private const val BOUNCE_TOP_POSITION = TOP_BORDER_POSITION - HEIGHT / 2f
+        private const val BOUNCE_BOTTOM_POSITION = -BOUNCE_TOP_POSITION
+        private const val BOUNCE_RIGHT_POSITION = RIGHT_BORDER_POSITION - WIDTH / 2f
+        private const val BOUNCE_LEFT_POSITION = -BOUNCE_RIGHT_POSITION
+
+        private fun getExpectedDirections(quadrant1: IntRange, quadrant2: IntRange, axis: Int):
+            MutableMap<Float, Float> {
+            val expectedDirections: MutableMap<Float, Float> = HashMap()
+            for (direction in -165..180 step 15) {
+                if (direction in quadrant1.union(quadrant2)) {
+                    if (direction >= 0) {
+                        expectedDirections[direction.toFloat()] = (axis - direction).toFloat()
```

Comment:
Instead of 
if (b) {
    x = y
} else {
    x = z
}
you can write 
x = if(b) {
   y
} else {
   z 
}
Or if there are more cases use 'when' 
https://kotlinlang.org/docs/reference/control-flow.html